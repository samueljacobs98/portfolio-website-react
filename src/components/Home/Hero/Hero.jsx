import "./Hero.scss";
import earth from "../../../assets/svgs/earth.svg";
const Hero = () => {
  return (
    <div className="hero">
      <h1 className="hero__heading">Hi, I'm Samuel</h1>
      <p className="hero__text">
        a software engineer based
        <br />
        in London, England
      </p>
      <img className="hero__earth" src={earth} alt="planet earth" />
    </div>
  );
};

export default Hero;
