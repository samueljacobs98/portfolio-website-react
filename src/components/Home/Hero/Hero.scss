@use "../../../assets/scss/variables" as *;

.hero {
  position: relative;
  &__heading {
    font-weight: normal;
    font-size: $font-size-big;
  }
  &__text {
    font-size: $font-size;
  }
  &__earth {
    position: absolute;
    bottom: -2ch;
    right: -0.5ch;
    animation-name: floating;
    animation-duration: 4s;
    animation-iteration-count: infinite;
    animation-timing-function: ease-in-out;
  }
}

@media screen and (max-width: 650px) {
  .hero {
    &__heading {
      font-size: 2rem;
    }
    &__text {
      font-size: 1rem;
    }
    &__earth {
      width: 50%;
      bottom: 1ch;
      animation-name: mob-floating;
    }
  }
}

@keyframes floating {
  0% {
    transform: translate(0, 0);
  }
  50% {
    transform: translate(0, -20px);
  }
  100% {
    transform: translate(0, 0);
  }
}

@keyframes mob-floating {
  0% {
    transform: translate(0, 0);
  }
  50% {
    transform: translate(0, -10px);
  }
  100% {
    transform: translate(0, 0);
  }
}
