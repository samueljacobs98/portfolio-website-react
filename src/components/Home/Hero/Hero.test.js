import { render } from "@testing-library/react";
import Hero from "./Hero";

test("should match snapshot", () => {
  const { container } = render(<Hero />);
  expect(container).toMatchSnapshot();
});
