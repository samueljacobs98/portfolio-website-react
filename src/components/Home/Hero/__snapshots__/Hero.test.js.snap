// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="hero"
  >
    <h1
      class="hero__heading"
    >
      Hi, I'm Samuel
    </h1>
    <p
      class="hero__text"
    >
      a software engineer based
      <br />
      in London, England
    </p>
    <img
      alt="planet earth"
      class="hero__earth"
      src="earth.svg"
    />
  </div>
</div>
`;
